CRVResponse/src/CrvWaveformsGenerator_module.cc

```diff
@@ -70,7 +70,14 @@ namespace mu2e
     std::vector<double> _digitizationPointShiftFEBsSide0, _digitizationPointShiftFEBsSide1;
     std::vector<double> _timeShiftFEBsSide0, _timeShiftFEBsSide1;
 
+    struct ChargeCluster
+    {
+      std::vector<double> times, charges;
```

Comment:
are times and charges are correlated vectors?  If so I suggest instead a vector  structs, where the struct is time, charge.  That is more efficient in terms of memory caching.

Reply:
Also, if there is a known maximum size for this vector consider using an array, or at least reserving the vector to that size.

Reply:
Yes, they are correlated. I can combine them into a struct or a pair. I could reserve the vector to the number of charges.